cmdLine/src/main/scala/topgun/cmdline/ClassLoaderInfo.scala

```diff
@@ -0,0 +1,32 @@
+package topgun.cmdline
+
+
+import java.io.InputStream
+
+import org.objectweb.asm.ClassReader
+import org.objectweb.asm.tree.ClassNode
+
+import scala.collection.mutable
+
+class ClassLoaderInfo() {
+  var allClasses = new mutable.HashMap[String, ClassInfo]()
+
+  def lookup(clazzName: String, clazzLoader: ClassLoader): ClassInfo = {
+    val key = clazzName.replace(".", "/") + ".class"
+    allClasses.getOrElseUpdate(clazzName, searchForAndPutClassInfoInToMap(key, clazzLoader)): ClassInfo
+  }
+
+  def searchForAndPutClassInfoInToMap(key: String, clazzLoader: ClassLoader): ClassInfo = {
+    val iStream: InputStream = clazzLoader.getResourceAsStream(key)
+    val reader: ClassReader = new ClassReader(iStream)
+    val classNode: ClassNode = new ClassNode()
+    reader.accept(classNode, 0)
+
+    val className: String = reader.getClassName
+    val sourceFile: String = classNode.sourceFile
+    val fileHierarchy: String = reader.getSuperName
```

Comment:
so this is the superclass, not the hierarchy ( which is a tree)
(we will also need the interfaces)

Reply:
okay, will add those.

Reply:
a little clarification on "hierarchy", are we looking for
1. a list of superclasses that bubble up to "java.lang.object"?